### PR TITLE
Make sure stratagem configuration is removed when fs is forgotten.

### LINKED ISSUE #1075: Add foreign key for filesystem in Stratagem Configuration model
Currently the stratagem configuration contains a reference to the filesystem id it is configured with. It does not, however, enforce referential integrity by using a foreign key. After discussing this further we cannot think of a case in which a stratagem configuration should remain after a filesystem is removed. Therefore, we should set the foreign key constraint on the filesystem_id column in the stratagem configuration table.

### LINKED ISSUE #1094: When forgetting the filesystem in Monitored mode, also forget Stratagem
When the user clicks the forget action on a monitored fs:

Is Stratagem is configured for the fs, it should have it's `UnconfigureStratagemJob` run as part of the fs removal. 

An open question is what should happen if someone attempts to remove the fs while a scan is occuring, but this can probably be handled in another issue.

### PR DESCRIPTION
**Pull request by johnsonw**
Fixes #1094.
- Forget stratagem configuration when forgetting in monitored mode
Fixes #1075.
- Add foregin key for filesystem in Stratagem Configuration model

Signed-off-by: Will Johnson <wjohnson@whamcloud.com>

### CODE REVIEW

**Comment by johnsonw** [comment]
Currently seeing the following error when the filesystem is removed:

```
panicked at 'Missing child el_ws', src/libcore/option.rs:1036:5

Stack:

Error
    at imports.wbg.__wbg_new_59cb74e423758ede (https://localhost:8443/wasm-components/package.js:769:21)
    at console_error_panic_hook::hook::h7c1d24eab36fab05 (wasm-function[740]:168)
    at core::ops::function::Fn::call::h81a88b70905367aa (wasm-function[5218]:3)
    at std::panicking::rust_panic_with_hook::h3f88564dfeb012eb (wasm-function[940]:259)
    at std::panicking::continue_panic_fmt::he2d0fc2a935878af (wasm-function[2496]:75)
    at rust_begin_unwind (wasm-function[5167]:3)
    at core::panicking::panic_fmt::hc4f83bfed80aeabd (wasm-function[2667]:70)
    at core::option::expect_failed::he602aa67d6d2d670 (wasm-function[2198]:90)
    at core::option::Option<T>::expect::hbaab7d75fb94eeac (wasm-function[3276]:33)
    at seed::vdom::patch::h7e5b3990b6ff31a6 (wasm-function[186]:2037)


imports.wbg.__wbg_error_4bb6c2a97407129a @ package.js:782
console_error_panic_hook::hook::h7c1d24eab36fab05 @ wasm-0082dd2e-740:151
core::ops::function::Fn::call::h81a88b70905367aa @ wasm-0082dd2e-5218:3
std::panicking::rust_panic_with_hook::h3f88564dfeb012eb @ wasm-0082dd2e-940:113
std::panicking::continue_panic_fmt::he2d0fc2a935878af @ wasm-0082dd2e-2496:35
rust_begin_unwind @ wasm-0082dd2e-5167:3
core::panicking::panic_fmt::hc4f83bfed80aeabd @ wasm-0082dd2e-2667:30
core::option::expect_failed::he602aa67d6d2d670 @ wasm-0082dd2e-2198:42
core::option::Option<T>::expect::hbaab7d75fb94eeac @ wasm-0082dd2e-3276:20
seed::vdom::patch::h7e5b3990b6ff31a6 @ wasm-0082dd2e-186:978
seed::vdom::App<Ms,Mdl,ElC>::rerender_vdom::h1b66abec4e81ed63 @ wasm-0082dd2e-283:313
seed::vdom::App<Ms,Mdl,ElC>::schedule_render::{{closure}}::hf856a29dd5c73d52 @ wasm-0082dd2e-2005:9
<dyn +core::ops::function::FnMut<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h1d6a6da5e864936c @ wasm-0082dd2e-3538:10
__wbg_elem_binding0 @ package.js:6
real @ package.js:1151
requestAnimationFrame (async)
imports.wbg.__widl_f_request_animation_frame_Window @ package.js:698
seed::util::request_animation_frame::ha1691c37861966ce @ wasm-0082dd2e-1737:12
seed::vdom::App<Ms,Mdl,ElC>::update::h2855dcc5bef40ab8 @ wasm-0082dd2e-250:362
fsdetailpagecallbacks_set_filesystem @ wasm-0082dd2e-643:192
set_filesystem @ package.js:338
(anonymous) @ file-system-detail-component.js:17
(anonymous) @ index.js:1799
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:2079
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:2079
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:2079
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:2403
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:2079
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
(anonymous) @ index.js:1609
(anonymous) @ broadcaster.js:31
(anonymous) @ broadcaster.js:31
(anonymous) @ index.js:1799
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:1730
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:2403
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:624
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
(anonymous) @ index.js:3875
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
(anonymous) @ index.js:1609
(anonymous) @ sse-handler.js:164
(anonymous) @ sse-handler.js:220
(anonymous) @ sse-handler.js:90
(anonymous) @ index.js:1799
i @ index.js:1530
k._send @ index.js:1091
k._writeOutgoing @ index.js:1384
_push_fn @ index.js:840
n.onmessage @ sse-stream.js:56
Show 56 more frames
22:44:56.146 wasm-0082dd2e-5458:2 Uncaught RuntimeError: unreachable
    at rust_panic (wasm-function[5458]:1)
    at std::panicking::rust_panic_with_hook::h3f88564dfeb012eb (wasm-function[940]:294)
    at std::panicking::continue_panic_fmt::he2d0fc2a935878af (wasm-function[2496]:75)
    at rust_begin_unwind (wasm-function[5167]:3)
    at core::panicking::panic_fmt::hc4f83bfed80aeabd (wasm-function[2667]:70)
    at core::option::expect_failed::he602aa67d6d2d670 (wasm-function[2198]:90)
    at core::option::Option<T>::expect::hbaab7d75fb94eeac (wasm-function[3276]:33)
    at seed::vdom::patch::h7e5b3990b6ff31a6 (wasm-function[186]:2037)
    at seed::vdom::App<Ms,Mdl,ElC>::rerender_vdom::h1b66abec4e81ed63 (wasm-function[283]:668)
    at seed::vdom::App<Ms,Mdl,ElC>::schedule_render::{{closure}}::hf856a29dd5c73d52 (wasm-function[2005]:16)
```

File: chroma_core/models/stratagem.py [hunk 1]
```diff
@@ -87,10 +112,10 @@ def run(self, kwargs):
                 "Description=Start Stratagem run on {}\n"
                 "\n[Timer]\n"
                 "OnBootSec={}\n"
-                "OnUnitActiveSec={}\n".format(config.filesystem_id, config.interval / 1000, config.interval / 1000)
+                "OnUnitActiveSec={}\n".format(config.filesystem.id, config.interval / 1000, config.interval / 1000)
```
Comment (utopiabound) [inline chroma_core/models/stratagem.py]: Can we switch config.filesystem.id to config.filesystem.name?  I know this preserves functionality, but it'll be nicer for end-user

File: chroma_core/models/stratagem.py [hunk 2]
```diff
@@ -87,10 +112,10 @@ def run(self, kwargs):
...
             )
 
-        iml_cmd = "/usr/bin/iml stratagem scan --filesystem {}".format(config.filesystem_id)
+        iml_cmd = "/usr/bin/iml stratagem scan --filesystem {}".format(config.filesystem.id)
```
Comment (utopiabound) [inline chroma_core/models/stratagem.py]: Same here: config.filesystem.name

File: chroma_core/models/stratagem.py [hunk 3]
```diff
@@ -104,7 +129,7 @@ def run(self, kwargs):
                 "After=iml-manager.target\n"
                 "\n[Service]\n"
                 "Type=oneshot\n"
-                "ExecStart={}\n".format(config.filesystem_id, iml_cmd)
+                "ExecStart={}\n".format(config.filesystem.id, iml_cmd)
```
Comment (utopiabound) [inline chroma_core/models/stratagem.py]: Again

**Pull request closed without merging**